src/ngram_matches.cc

```diff
@@ -44,14 +44,13 @@ namespace fuzzy
     return _psentences.size();
   }
 
-  std::vector<unsigned>
-  NGramMatches::sentence(int s_id) const
+  const unsigned*
+  NGramMatches::sentence(unsigned s_id, unsigned* length) const
```

Comment:
Maybe you should introduce a `SuffixString` (or maybe with a better name), to group the return value and `length`.